src/Entity/DecoupledAuthUser.php

```diff
@@ -54,6 +54,9 @@ public function isDecoupled() {
    * {@inheritdoc}
    */
   public function setDecoupled($decoupled = NULL) {
+    if (is_null($decoupled) && $this->isNew() && $this->name->value === NULL) {
+      $this->name->value = '';
+    }
     $this->decoupled = $decoupled ?: $this->name->value === NULL;
```

Comment:
There is probably a better way of doing this, but I'm not sure what it is... Basically all users on the user register form were defaulting to decoupled because $this->name->value is NULL when the form is built:
```php
$user_storage->create([]);
```